Update starting-builds-automatically.md (#1803)
codemagic-ci-cd/codemagic-docs: content/yaml-running-builds/starting-builds-automatically.md

Please refer to [Wildcard Match Documentation](https://facelessuser.github.io/wcmatch/fnmatch/) for more advanced matching patterns.

## Working with Pull Requests

> When dealing with Pull Requests, you have two options: you can either focus on the branch where the proposed changes are made, or you can target the destination branch after the PR has been merged.

Reviewer: ```suggestion
When dealing with pull requests, you have two options: you can either focus on the branch where the proposed changes are made, or you can target the destination branch after the pull request has been merged.
```

Example 1 - When creating pull requests on the `feature` branch, remember to set `source:false`. This will ensure that the build runs on the proposed code changes within the `feature` branch.

{{< highlight yaml "style=paraiso-dark">}}
triggering:
      events:
        - pull_request
      branch_patterns:
        - pattern: 'feature'
          include: true
          source: false  
{{< /highlight >}}

> Example 2 - To merge Pull Requests with the `master/main` branch, simply set `source:true`. This will trigger the build on `master/main` branch once the pull request has been merged.

Reviewer: ```suggestion
Example 2 - To merge pull requests with the `master/main` branch, simply set `source:true`. This will trigger the build on `master/main` branch once the pull request has been merged.
```

Please refer to [Wildcard Match Documentation](https://facelessuser.github.io/wcmatch/fnmatch/) for more advanced matching patterns.

## Working with Pull Requests
When dealing with pull requests, you have two options: you can either focus on the branch where the proposed changes are made, or you can target the destination branch after the pull request has been merged.

> Example 1 - When creating pull requests on the `main` branch from, for, e.g. `feature` branch, which is a way to propose and review changes before they're integrated into `main`, remember to set `source:false` and `patter:main`. This will ensure that the build runs on the proposed code changes within the `feature` branch when pull request is created or updated.

Reviewer: ```suggestion
Example 1 - When creating pull requests on the `main` branch from a `feature` branch, which is a way to propose and review changes before they're integrated into `main`, remember to set `source:false` and `pattern:main`. This will ensure that the build runs on the proposed code changes within the `feature` branch when pull request is created or updated.
```

{{< highlight yaml "style=paraiso-dark">}}
triggering:
      events:
        - pull_request
      branch_patterns:
        - pattern: 'main'
          include: true
          source: false  
{{< /highlight >}}

> Example 2 - Setting `source:true` and `branch:main`, will trigger the build on the `main` branch once the pull request has been merged from for, e.g. `feature` branch into the `main` branch.

Reviewer: ```suggestion
Example 2 - Setting `source:true` and `branch:main`, will trigger the build on the `main` branch once the pull request has been merged from the `feature` branch into the `main` branch.
```
